addons/website/static/src/systray_items/translate_website.js

```diff
@@ -6,10 +6,39 @@ import { Component, useState } from "@odoo/owl";
 
 class TranslateWebsiteSystray extends Component {
     setup() {
+        this.rpc = useService("rpc"),
         this.websiteService = useService('website');
         this.websiteContext = useState(this.websiteService.context);
     }
 
+    attemptStartTranslate() {
+        if (this.websiteService.isRestrictedEditor && !this.websiteService.isDesigner) {
+            const object = this.websiteService.currentWebsite.metadata.mainObject;
+            const others = {};
+            const otherRecordEls = this.websiteService.websiteRootInstance.el.querySelectorAll("[data-res-model][data-res-id]:not([data-res-model='ir.ui.view'])");
+            for (const el of otherRecordEls) {
+                if (!others[el.dataset.resModel]) {
+                    // Keep one record of each type.
+                    others[el.dataset.resModel] = parseInt(el.dataset.resId);
+                }
+            }
+            this.rpc('/website/check_can_modify_any', {
+                res_model: object.model,
+                res_id: object.id,
+                others: Object.entries(others).map((kv) => {
+                    return {
+                        res_model: kv[0],
+                        res_id: kv[1],
+                    };
+                }),
+            }).then(() => {
+                this.startTranslate();
+            });
```

Comment:
```suggestion
            const objects = {};
            objects[object.model] = object.id;
            const otherRecordEls = this.websiteService.websiteRootInstance.el.querySelectorAll("[data-res-model][data-res-id]:not([data-res-model='ir.ui.view'])");
            for (const el of otherRecordEls) {
                if (!objects[el.dataset.resModel]) {
                    // Keep one record of each type.
                    objects[el.dataset.resModel] = parseInt(el.dataset.resId);
                }
            }
            this.rpc('/website/check_can_modify_any', {
                records: Object.entries(objects).map((kv) => {
                    return {
                        res_model: kv[0],
                        res_id: kv[1],
                    };
                }),
            }).then(() => {
                this.startTranslate();
            });
```

Reply:
@JKE-be I pushed the suggested route signature change